commands/service_execute.go

```diff
@@ -141,15 +141,20 @@ func (c *serviceExecuteCmd) getData(taskKey string, s *coreapi.Service, dataStru
 		return string(b), err
 	}
 
-	if c.jsonFile == "" {
-		if survey.AskOne(&survey.Input{Message: "Enter the filepath to the inputs"}, &c.jsonFile, nil) != nil {
-			return "", errors.New("no filepath given")
+	// see if task has no inputs.
+	for _, task := range s.Tasks {
+		if task.Key == taskKey {
```

Comment:
I don't quite like extending types that are generated by protobuf. /cc @antho1404 and  @NicolasMahe what do you think about this?

We can also create a func inside this file to get tasks by name from _Service_ type. But here we use it just for once, do you think we need it? If so, I can do the changes.

Reply:
> I don't quite like extending types that are generated by protobuf.

Why not? it's just a type, otherwise, we can create another type that like
```go
type MyTask struct {
  coreapi.Task
}

// all metods go here
```

Or as you said function somewhere in the file.

1) BAD - creating another type and extending it is not so clean and easy 
2) BAD - creating functions in some packages and you will end up with functions everywhere - in one package for handling Task, in other for handling Parameter, Service etc...

3) Extend type where it was generated - What are cons of this? Pros - you have ever method related to those types in one place.

Reply:
For example, protobuf generates _GetTasks()_ method already and in future if they introduce a method like _GetTask()_ or something like that and if we do create the same method today as well by extending _Service_ type, then we'll have a conflict. I know that this is a rare case but I don't like to mess with types that are auto-generated because we don't have a full control on them.

But, I agree that if we can keep this kinda helper funcs in one place like _coreapi_ pkg, this way we'll not need to duplicate them in other pkgs. In that case I'd like to have pkg level funcs instead of methods so for example something like this: `coreapi.GetTask(s *Service, taskName string)`

Reply:
Let's keep it simple for now. The command doesn't import the package `service` and I like to keep it this way.

Let's think about it in a refactor issue / PR